cmds/bmp/update_state_command.go

```diff
@@ -45,10 +47,81 @@ func (cmd updateStateCommand) Options() cmds.Options {
 
 func (cmd updateStateCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if cmd.options.Server == "" {
+		return false, errors.New("cannot have empty server ID")
+	}
+
+	if cmd.options.State == "" {
+		return false, errors.New("cannot have empty state")
+	} else {
+		if !isValidState(cmd.options.State) {
+			return false, errors.New("the server state is incorrect!")
+		}
+	}
+
 	return true, nil
 }
 
 func (cmd updateStateCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s command: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
-	return 0, nil
+
+	cmd.ui.PrintlnInfo("Be careful to update the state of a server, which might cause Bluemix broken!")
+	if isConfirmed(cmd) {
+		updateStatusResponse, err := cmd.bmpClient.UpdateStatus(cmd.options.Server, cmd.options.State)
+		if err != nil {
+			return 1, err
+		}
+
+		if updateStatusResponse.Status != 200 {
+			return updateStatusResponse.Status, nil
+		}
+
+		cmd.ui.PrintlnInfo("Update Successful!")
+
+		return 0, nil
+	} else {
+		return 1, errors.New("Update Cancelled!")
+	}
+}
+
+// Private Methods
```

Comment:
Why are these private functions and not private methods? I generally like to have them be private methods, e.g., 

```
func (cmd updateStateCommand) isValidState(state string) bool {
...
}
...
```

That way it's scoped to this class. If the method is useful other places then move to a global function in the `common` package.